src/loader.js

```diff
@@ -146,15 +147,186 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
+    if (__global.Map)
+      registry._registry.registryMap = new __global.Map();
+    else
+      registry._registry.registryMap = {};
+    // 4.4.2
+    if (__global.Symbol && __global.Symbol.iterator)
+      registry[__global.Symbol.iterator] = RegistryPrototype.prototype.entries;
+    return registry;
+  }
+
+  // 4.2
+  function Registry() {
+    throw new Error('A registry may only be created when creating a loader');
+  }
+
+  // 4.3.1
+  Registry.prototype = RegistryPrototype;
+
+  // 4.4 (reason for a separate constructor explained at 4.3.1)
+  function RegistryPrototype() {}
+
+  // 4.4.1
+  Registry.prototype.constructor = Registry;
+
+  // 4.4.2 is inlined in 4.1.1
+
+  // 4.4.3
+  RegistryPrototype.prototype.entries = function() {
+    if (typeof this !== 'object')
+      throw new TypeError('cannot get entries of a non-registry');
+    if (__global.Map) {
+      // native iterator
+      return this._registry.registryMap.entries();
+    } else if (__global.Symbol && __global.Symbol.iterator) {
```

Comment:
If the user is using a polyfill for iterator that does not overwrite the global `Symbol.iterator`, surely it should still be possible for them to use this `entries` feature though? That is, do we really need this check?